src/test/java/com/puppycrawl/tools/checkstyle/internal/XdocsPagesTest.java

```diff
@@ -1885,89 +1901,80 @@ private static void validateChapterWiseTesting(Iterator<Node> itrChecks,
             Node config = null;
 
             try {
-                config = itrConfigs.next();
+                config = itrChecks.next();
             }
             catch (NoSuchElementException ignore) {
                 assertWithMessage(styleName + "_style.xml rule '" + ruleName + "' module '"
-                        + moduleName + "' is missing the config link: config").fail();
+                        + moduleName + "' is missing the config link").fail();
             }
 
             final String configUrl = config.getAttributes().getNamedItem("href")
                     .getTextContent();
 
-            final String expectedUrl = "https://github.com/search?q="
-                    + "path%3Asrc%2Fmain%2Fresources%20path%3A**%2F" + styleName
+            final String wholeConfigUrl = partialConfigUrl
                     + "_checks.xml+repo%3Acheckstyle%2Fcheckstyle+" + moduleName;
 
             assertWithMessage("google_style.xml rule '" + ruleName + "' module '"
                     + moduleName + "' should have matching config url")
                     .that(configUrl)
-                    .isEqualTo(expectedUrl);
-
+                    .isEqualTo(wholeConfigUrl);
         }
 
-        if (itrConfigs.hasNext()) {
+        if (itrSample.hasNext()) {
             assertWithMessage(styleName + "_style.xml rule '" + ruleName + "' should have checks"
-                    + " if it has config/test links")
+                    + " if it has check/config links")
```

Comment:
`if it has check/config links` => `if it has sample links`.

You are checking samples here. Not configs.

Reply:
done